src/libm/CMakeLists.txt

```diff
@@ -1,31 +1,66 @@
+include_directories(${CMAKE_CURRENT_BINARY_DIR}/include/)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+
 # Helper executable: generates parts of the sleef header file
 add_executable(${TARGET_MKRENAME} mkrename.c)
-target_compile_options(${TARGET_MKRENAME}
-  PRIVATE ${FLAGS_WALL})
 target_compile_features(${TARGET_MKRENAME}
   PUBLIC c_variadic_macros)
+# Helper executable: dispatcher for the vector extensions
+add_executable(${TARGET_MKDISP} mkdisp.c)
+target_compile_features(${TARGET_MKDISP}
+  PUBLIC c_variadic_macros)
+
+set(RENAME_HEADERS "")
+set(TARGET_OBJECTS "")
+
+# Single precision and double precision
+set(SIMD_SOURCES sleefsimdsp.c sleefsimddp.c)
+
+if(COMPILER_SUPPORTS_SSE2)
+  set(HEADER_SSE2 ${CMAKE_CURRENT_BINARY_DIR}/include/renamesse2.h)
+  set(OBJECT_SSE2 "sleefsse2")
+
+  add_library(${OBJECT_SSE2} OBJECT ${SIMD_SOURCES} ${HEADER_SSE2})
```

Comment:
I am afraid we are doing something wrong here, especially because it requires to specify `-fPIC` as `target_compile_option`. In [1], the OBJECT version of `add_library` is described as: 

>  Object libraries may contain only sources that compile, header files, and other files that would not affect linking of a normal library

For now let's keep it this way, but if it turn out that we need to determine which `-fPIC` we need for each compiler/OS we want to support, I think we will have to rethink this part. Please add a comment here stating this problem.

[1] https://cmake.org/cmake/help/v3.4/command/add_library.html#object-libraries